client/src/extension.ts

```diff
@@ -5,46 +5,145 @@
 
 import * as fs from "fs";
 import * as path from "path";
-import { promisify } from "util";
-import { workspace, ExtensionContext, commands } from "vscode";
-
-let exists = promisify(fs.exists);
+import { window, ProgressLocation, workspace, ExtensionContext, commands } from "vscode";
+import * as semver from 'semver';
 
 import {
   LanguageClient,
   LanguageClientOptions,
   ServerOptions,
-  TransportKind
 } from "vscode-languageclient";
 
+import { log, run, options } from './utils';
+import { getEsyStatus } from './esy-utils';
+import { ToolNotFound, OpamNotFound, EsyNotFound, OpamOnWindowsError } from './errors';
+import { promisify } from "util";
+
 let client: LanguageClient;
 
-async function isEsyProject() {
+async function getCommandForWorkspace(context): Promise<{ command: string, args: string[] }> {
+
   let root = workspace.rootPath;
-  if (await exists(path.join(root, "package.json"))) {
-    return true;
-  } else if (await exists(path.join(root, "esy.json"))) {
-    return true;
-  } else {
-    return false;
-  }
-}
+  let esyStatus = await getEsyStatus(root);

```

Comment:
Do we need to check if esy is installed?

Reply:
Indeed, we should. I presumed I did already, must have removed at the last moment.

Reply:
Silly me. Just remembered,  esy is bundled. I think a PATH issue isnt making esy available yet. So, with that out of the way, we wont need to check if esy is installed